src/material/progress-bar/progress-bar.ts

```diff
@@ -79,12 +79,9 @@ export class MatProgressBar
     super(elementRef);
     this._isNoopAnimation = _animationMode === 'NoopAnimations';
 
-    if (defaults) {
-      if (defaults.color) {
+    if (defaults && defaults.color) {
```

Comment:
This is not equivalent to the current code. This will not set the `mode` if `color` is not provided.